README.md

```diff
@@ -110,16 +87,39 @@ In this task, you will need to write and run a Powershell script, which deploys
             sudo systemctl enable todoapp
         ```
     
-    4. Verify that web app service is running, for that, run the following command on the VM: 
+    4. Verify that the web app service is running; for that, run the following command on the VM: 
         
         ```
             systemctl status todoapp
         ```
 
+## How To Complete Tasks in This Module 
+
+Tasks in this module are relying on 2 PowerShell scripts: 
+
+- `scripts/generate-artifacts.ps1` generates the task  “artifacts” and uploads them to cloud storage. An  “artifact” is evidence of a task completed by you. Each task will have its own script, which will gather the required artifacts. The script also adds a link to the generated artifact in the `artifacts.json` file in this repository — make sure to commit changes to this file after you run the script. 
+- `scripts/validate-artifacts.ps1` validates the artifacts generated by the first script. It loads information about the task artifacts from the `artifacts.json` file.
+
+Here is how to complete tasks in this module:
+
+1. Clone task repository
+
+2. Make sure you completed the steps described in the **Prerequisites** section
+
+3. Complete the task, described in the **Requirements** section 
+
+4. Run `scripts/generate-artifacts.ps1` to generate task artifacts. Script will update the file `artifacts.json` in this repo. 
+
+5. Run `scripts/validate-artifacts.ps1` to test yourself. If tests are failing — follow the recomendation from the test script error message to fix or re-deploy your infrastructure. When you will be ready to test yourself again — **re-generate the artifacts** (step 4) and re-run tests again. 
+
+6. When all tests will pass — commit your changes and submit the solution for a review. 
+
+Pro tip: if you stuck with any of the implementation steps — run `scripts/generate-artifacts.ps1` and `scripts/validate-artifacts.ps1`. The validation script might give you a hint on what you should do.  
+
 4. Run artifacts generation script `scripts/generate-artifacts.ps1`
 
 5. Test yourself using the script `scripts/validate-artifacts.ps1`
 
 6. Make sure that changes to both `task.ps1` and `result.json` are commited to the repo, and sumbit the solution for a review.
 
-7. When solution is validated, - delete all resources you deployed with the script, they won't be used in the next tasks.  
+7. When solution is validated, — delete all resources you deployed with the script, they won't be used in the next tasks.  
```

Comment:
Step 7 belongs to this particular task - it should be the last step in the "Requirements" section.